devicetypes/zooz/zooz-zse44-temperature-humidity-xs-sensor.src/zooz-zse44-temperature-humidity-xs-sensor.groovy

```diff
@@ -0,0 +1,418 @@
+	lowHumidityThreshold: [num:11, title:"Low Humidity Alert Level", size:1, defaultVal:0, range:"0..100", desc:"0(disabled), 1..100(%)"]
+]
+
+@Field static Map temperatureSensor = [sensorType:1, scale:1]
+@Field static Map humiditySensor = [sensorType: 5, scale:0]
+@Field static Map temperatureAlarm = [name:"temperatureAlarm", notificationType:4, normalEvent:0, highEvent:2, lowEvent:6]
+@Field static Map humidityAlarm = [name:"humidityAlarm", notificationType:16, normalEvent:0, highEvent:2, lowEvent:6]
+@Field static int wakeUpInterval = 43200
+
+metadata {
+	definition (
+		name: "Zooz ZSE44 Temperature | Humidity XS Sensor",
+		namespace: "Zooz",
+		author: "Kevin LaFramboise (@krlaframboise)",
+		ocfDeviceType:"oic.d.thermostat",
+		vid: "064b47e2-afde-3228-89a4-480f0f55b3fd",
+		mnmn: "SmartThingsCommunity"
+	) {
+		capability "Sensor"
+		capability "Temperature Measurement"
+		capability "Relative Humidity Measurement"
+		capability "Battery"
+		capability "Refresh"
+		capability "Health Check"
+		capability "Configuration"
+		capability "platemusic11009.temperatureHumiditySensor"
+		capability "platemusic11009.temperatureAlarm"
+		capability "platemusic11009.humidityAlarm"
+		capability "platemusic11009.firmware"
+		capability "platemusic11009.syncStatus"
+
+		// zw:Ss2a type:0701 mfr:027A prod:7000 model:E004 ver:1.10 zwv:7.13 lib:03 cc:5E,55,9F,6C sec:86,85,8E,59,31,72,5A,87,73,80,71,70,84,7A
+		fingerprint mfr:"027A", prod:"7000", model:"E004", deviceJoinName: "Zooz Temperature Humidity Sensor" // Zooz ZSE44 Temperature | Humidity XS Sensor
+	}
+
+	preferences {
+		configParams.each { name, param ->
+			if (param.options) {
+				input name, "enum",
+					title: param.title,
+					description: "Default: ${param.options[param.defaultVal]}",
+					required: false,
+					displayDuringSetup: false,
+					defaultValue: param.defaultVal,
+					options: param.options
+			} else if (param.range) {
+				input name, "number",
+					title: param.title,
+					description: "${param.desc} - Default: ${param.defaultVal}",
+					required: false,
+					displayDuringSetup: false,
+					defaultValue: param.defaultVal,
+					range: param.range
+			}
+		}
+
+		input "debugLogging", "enum",
+			title: "Logging:",
+			description: "Default: Enabled",
+			required: false,
+			defaultValue: "1",
+			options: ["0":"Disabled", "1":"Enabled"]
+	}
+}
+
+def installed() {
+	logDebug "installed()..."
+	state.pendingRefresh = true
+	initialize()
+}
+
+def updated() {
+	logDebug "updated()..."
+	initialize()
+
+	if (pendingChanges) {
+		logForceWakeupMessage("The setting changes will be sent to the device the next time it wakes up.")
+	}
+}
+
+void initialize() {
+	state.debugLoggingEnabled = (safeToInt(settings?.debugLogging, 1) != 0)
+
+	refreshSyncStatus()
+
+	if (device.currentValue("temperatureHumidity") == null) {
+		state.displayHumidity = " "
+		state.displayTemperature = " "
```

Comment:
Please, could You explain why do You initialize those state variables with a whitespace character ?

Reply:
It was the simplest workaround to a UI issue I ran into where the dashboard would only display either the temp or humidity even though that custom attribute had both values in it.  

I can't if that was happening with the Android or iOS mobile app that, but those lines have no negative impact on the readability or functionality of the DTH so I don't see any reason to change them.